sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -132,6 +132,7 @@ case class In(value: Expression, list: Seq[Expression]) extends Predicate
   }
 
   override def children: Seq[Expression] = value +: list
+  lazy val inSetConvertible = children.forall(_.deterministic)
```

Comment:
Oh, then, the semantic is different. What you mean is just improving `InSet`.
My original PR was about for deletion about all deterministic duplications.